Extract Stats Remote Services into WordPressKit dynamic framework

astralbodies opened the pull request:
Issue #7144 

This PR merges the results of PR #7146 plus extraction of the stats remote services into WordPressKit. WordPressKit is a dynamic framework that will eventually house all remote services and related data models. This is the first major re-structuring of the networking stack to support the iOS networking rearchitecture project.

To test:
1. Close Xcode.
2. Delete your derived data (to be safe).
3. `pod install`.
4. Open the workspace.
5. Build & run unit tests.
6. Build & run the app. Make sure stats works properly.

Needs review: @elibud, @frosty, @bummytime

frosty commented:
I think now we can remove all the framework-style imports for `WordPressKit` headers, and instead switch to a single module import wherever we need it. i.e. take out all of the imports such as:

```
#import <WordPressKit/StatsItemAction.h>
```

And just switch to:

```
@import WordPressKit;
```

frosty commented:
Other than the imports mentioned above, everything else looks good. Builds, tests, and runs just fine :)

astralbodies commented:
@elibud Addressed your comment and removed the CLANG_ENABLE_MODULES override in the framework target (stupid Xcode!). @frosty Addresses your comment and removed all imports of specific WordPressKit headers and opted for a module import. On the files I touched if I saw other imports (like from shared or analytics) I fixed those too. 

Also resolved the merge conflict from develop. I shall merge this PR when Buddybuild gives me a green light.

Thanks everyone for taking the time to review this!
